src/foam/graphics/TreeNode.js

```diff
@@ -0,0 +1,297 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.graphics',
+  name: 'TreeNode',
+  extends: 'foam.graphics.Box',
+
+  requires: [
+    'foam.graphics.Box',
+    'foam.graphics.CView',
+    'foam.graphics.Label',
+    'foam.graphics.Line'
+  ],
+
+  imports: [
+    'formatNode',
+    'graph',
+    'nodeHeight',
+    'nodeWidth',
+    'padding',
+    'parentNode?',
+    'relationship'
+  ],
+  exports: ['as parentNode'],
+
+  properties: [
+    'data',
+    {
+      name: 'outline',
+      expression: function (x, nodeWidth, expanded, childNodes, padding) {
+        var nodeLeftEdgePadded = x - nodeWidth / 2 - padding / 2;
+        var nodeRightEdgePadded = x + nodeWidth / 2 + padding / 2;
+        var rootLevelOutline = [
+          {
+            left: nodeLeftEdgePadded,
+            right: nodeRightEdgePadded
+          }
+        ];
+
+        var champion = [];
```

Comment:
This isn't a very descriptive name for somebody looking at this code for the first time. It's more like a `outlinePerLevel` or `levelEdges` or something. Or maybe even just `returnValue` or `outline`.

Reply:
Renamed to outlineBelowRoot as not to confuse with the childOutline variable.